### PR TITLE
Iteration 2.1

### PR DESCRIPTION
**Pull request by connorpoland**
Scheduler - updated to specify items being manipulated in the queues as pickups or drop offs
ElevatorQueue - keeps pickups and drop offs seperate so the scheduler can better inform the floor subsystem. This will be helpful in iter3.
ElevatorSubSystem - ArrivalSensor has been in the elevator subsystem spec for a while now, finally got around to moving it in. UML will need updating.
FloorSubsystem - removed the arrival sensors and reworked how the direction lamps are used and updated based on the schedulers intention.

Basically just prepping the system for iteration 3.

### CODE REVIEW

File: src/elevator/ElevatorQueue.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		return dropOffQueue.peek();
+	}
+	
+	public Integer remove() {
+		int peek = peek();
+		if (peek != -1) {
+			while (peek == pickUpPeek()) {
+				pickUpQueue.remove();
+			}
+			while (peek == dropOffPeek()) {
```
Comment (JackyChiu) [inline src/elevator/ElevatorQueue.java]: you don't actually need the `while` functionality here, right? An`if` statement would work instead?

File: src/elevator/ElevatorSubSystem.java [hunk 2]
```diff
@@ -90,24 +95,12 @@ public void handleMessage(Message m) {
...
 
 	private void sendTravelMessage() {
-		FloorTravelMessage m = new FloorTravelMessage();
-		m.setElevator(elevator.getId());
-		m.setStartingFloor(elevator.getFloor());
-
-		int diff = 1;
 		if (elevator.getState() == Elevator.State.MOVING_UP) {
-			m.setDirection(Direction.UP);
+			arrivalSensor = new Thread(new ArrivalSensor(elevator.getId(), elevator.getFloor(), elevator.getFloor()+1));
```
Comment (kirinrastogi) [inline src/elevator/ElevatorSubSystem.java]: Can you explain why this is creating a new arrivalSensor instead of creating a new message and sending it to the existing sensor?

File: src/elevator/ElevatorQueue.java (hunk 1 above)
Comment (connorpoland) [reply src/elevator/ElevatorQueue.java]: This is intended to account for multiple of the same pick up and drop off (if the queue looks like {1, 2, 2, 2, 3, 5} for example).

File: src/elevator/ElevatorSubSystem.java (hunk 2 above)
Comment (connorpoland) [reply src/elevator/ElevatorSubSystem.java]: The arrivalSensor thread terminates once it sends the arrival message; its not like server. I'm open to changing this though.
